docs/spec_draft.md

```diff
@@ -1243,6 +1244,78 @@ Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.slice
+
+### Semantics
+
+Extracts a sub-tensor from the `operand` using the dimension indices from the
+operands.
+
+More formally, `result[i0, ..., ik, ..., iR-1] = operand[j0+s0, ..., jk+sk, ..., jR-1+sR-1]`
```

Comment:
Thanks! I've been wondering how to deal with the first element since stride does not apply to it.

Reply:
I don't think the proposed formula is correct. Stride means skipping X elements at a time, not shifting everything by X elements.  Here's how the correct formula could look like:

```
More formally, `result[i0, ..., iR-1] = operand[j0, ..., jR-1]` where
`jd = start_indices[d] + id * strides[d]`.
```

A few additional notes about this formula in comparison with the wording in the pull request and proposals:
  * Instead of `i0, ..., ik, ..., iR-1`, I think we can say `i0, ..., iR-1`. All dimensions are treated the same by this op, so there's no need to call out `ik`.
  * I think that "where `jd = ...`" is easier to reach than when the `jd` formula is replicated inline.
  * Stuff like `jd <= limit_indices[d]` and `d < R` follows from constraints, so I don't think we should mention it here.
  * I think that using `d` here is a better fit than `k` because constraints use `d`.

Overall, after we're done with Data Movement and Reduction ops, we should review the notation of all ops and make sure it's consistent: #289. I don't think we need to do this now because it is likely that future ops will uncover further notions and potential inconsistencies.